tests/test_permissions_synchronizer.py

```diff
@@ -667,6 +667,97 @@ def test_webhooks_invalid_service(self):
+        """
+        self.data["sync_permissions"] = {
+            "user_workspace": {
+                "services": {
+                    ServiceTHREDDS.service_type: {
+                        "Thredds_file_src": [
+                            {"name": self.test_service_name, "type": Service.resource_type_name},
+                            {"name": "private", "type": Directory.resource_type_name},
+                            {"name": "directory1", "type": Directory.resource_type_name},
+                            {"field": "resource_display_name",
+                             "regex": r"[\w]+:[\w\/.-]+",
+                             "type": File.resource_type_name}],
+                        "Thredds_file_target": [
+                            {"name": self.test_service_name, "type": Service.resource_type_name},
+                            {"regex": r"(?<=catalog:)[\w\/]*\/?(?=\/)", "type": Directory.resource_type_name},
+                            {"regex": r"[^\/]+$", "type": File.resource_type_name}],
+                        "Thredds_dir_src": [
+                            {"name": self.test_service_name, "type": Service.resource_type_name},
+                            {"name": "private", "type": Directory.resource_type_name},
+                            {"field": "resource_display_name", "regex": r"[\w]+:[\w\/]+",
+                             "type": Directory.resource_type_name}],
+                        "Thredds_dir_target": [
+                            {"name": self.test_service_name, "type": Service.resource_type_name},
+                            {"regex": r"(?<=:)[\w\/]+", "type": Directory.resource_type_name}]}},
+                "permissions_mapping": [
+                    f"Thredds_file_src : {Permission.READ.value} -> "
+                    f"Thredds_file_target : {Permission.READ.value}",
+                    f"Thredds_dir_src : {Permission.READ.value} -> "
+                    f"Thredds_dir_target : {Permission.READ.value}",
+                ]
+            }
+        }
+        with open(self.cfg_file.name, mode="w", encoding="utf-8") as f:
+            f.write(yaml.safe_dump(self.data))
+        app = utils.get_test_app(settings={"cowbird.config_path": self.cfg_file.name})
+        # Recreate new magpie handler instance with new config
+        HandlerFactory().create_handler("Magpie")
+
+        with contextlib.ExitStack() as stack:
+            stack.enter_context(mock.patch("cowbird.handlers.impl.thredds.Thredds",
+                                           side_effect=utils.MockAnyHandler))
+
+            # Create test resources
+            dir_src_res_id = self.magpie.create_resource("private", Directory.resource_type_name, self.test_service_id)
+            parent_res_id = self.magpie.create_resource("directory", Directory.resource_type_name, dir_src_res_id,
+                                                        f"{self.test_service_name}:directory")
+            dir1_src_res_id = parent_res_id
```

Comment:
I wonder if the names could be different or more precise to avoid confusion. 
`dir1_src_res_id` (`directory`) vs `dir_src_res_id`(`private`)

Reply:
`dir_src_res_id `(directory) and `private_src_res_id` (private) ?

Reply:
Works for me!